### ISSUE TITLE
Intermittent compile failures (invalid memory access)

### ISSUE THREAD

**Issue by kimburgess** [PlaceOS/rest-api#88]
There is an intermittent issue impacting the nightly builds:
```
#10 [stage-0  8/12] RUN UNAME_AT_COMPILE_TIME=true     PLACE_COMMIT=f13e79a7...
#10 83.68 Invalid memory access (signal 11) at address 0x7f115f427198
#10 83.68 [0x7f119c045f96] ???
#10 83.68 [0x7f119bf8fbeb] ???
#10 83.68 [0x7f119ce13831] ???
```
<sup>[full logs](https://github.com/PlaceOS/release/runs/1645524190?check_suite_focus=true)</sup>

This is most likely a compiler bug, but seems to only be occurring on this service.

Possible [related issues](https://github.com/crystal-lang/crystal/issues?q=is%3Aissue+is%3Aopen+%22invalid+memory+access%22+OR+%22signal+11%22).

**Comment by caspiano** [comment]
@kimburgess Are you still seeing this behaviour?

**Comment by kimburgess** [comment]
Last occurrence in the nightlies that impacted this service was [build 126](https://github.com/PlaceOS/release/runs/1714486649?check_suite_focus=true).

Looking at the [diff since then](https://github.com/PlaceOS/rest-api/compare/73d238720608cc4a87ded937b6bdca2fe7302628..9bfbfb4) likely resolved with the bump to crystal 0.36.1.
